normalize data for dcm

Eh2406 opened the pull request:
This is a start on https://github.com/UDST/urbansim/issues/208 still to do:

- [] get ci green
- [] add l1 l2 regularization
- [] test regularization

But I thout I'd open this to get some feedback.

Eh2406 commented:
AppVeyor is `Failed building wheel for pandana` I don't think I can fix that.
I am working on the travis things.

janowicz commented:
Thanks for this @Eh2406!   The passing Travis builds are the relevant benchmark here.  The appveyor Windows build are failing due to pandana on Windows, which is unrelated to the present PR.   We'll fix that, or we may remove pandana as a hard dependency of core urbansim (as little in this repo specifically requires pandana).

Eh2406 commented:
Ping. To be clear I am stuck waiting for advice on what tests you would like to see.

janowicz commented:
Some thoughts:
`test_dcm.py`
- Create a "normalized_dcm" fixture that takes the existing basic_dcm fixture as a starting point and set values for normalized/l1/l2
- Using the test_mnl_dcm function as a template, implement a high-level smoke-test just to see that the model with normalization can be fitted, that probabilities can be calculated from the resulting model, and that  'Normalization Mean' / 'Normalization Std' are in the resulting fit parameters.

`test_mnl.py`
- Can any of the normalization/regularization changes to mnl.py be tested here?  Perhaps a test for expected coefficients that are different when non-zero l1/l2 values supplied?  And a smoke-test to ensure that mnl.mnl_simulate() runs with normalization_mean / normalization_std?

Eh2406 commented:
I parameterized `basic_dcm` to run with normalization and 3 different regularizations. Then looked through all the newly failing tests. This sussed out 2 places where I did not pass the new arguments in the correct order. Oops. Meny of the test can be made to pass with an if on `basic_dcm.normalize`. For the 3 that could not be done so easily, I added a `simple_dcm` that matched the old `basic_dcm` and switched them to use that.

Thoughts?

Eh2406 commented:
merged and ci is green. Thoughts? What else would you like to see before accepting?
